src/service/cid-impl.js

```diff
@@ -182,15 +182,15 @@ function getOrCreateCookie(cid, getCidStruct, persistenceConsent) {
   const newCookie = 'amp-' + cid.sha384Base64_(getEntropy(win));
 
   // Store it as a cookie based on the persistence consent.
-  persistenceConsent.then(() => {
+  return persistenceConsent.then(() => {
     // The initial CID generation is inherently racy. First one that gets
     // consent wins.
     const relookup = getCookie(win, scope);
     if (!relookup) {
       setCidCookie(win, scope, newCookie);
     }
+    return Promise.resolve(relookup || newCookie);
```

Comment:
ack. Letting the tests run to make sure nothing is broken. Will fix it after that.